internal/component/poddistruptionbudget/poddisruptionbudget.go

```diff
@@ -107,6 +113,11 @@ func buildResource(etcd *druidv1alpha1.Etcd, pdb *policyv1.PodDisruptionBudget)
 	pdb.Spec.Selector = &metav1.LabelSelector{
 		MatchLabels: druidv1alpha1.GetDefaultLabels(etcd.ObjectMeta),
 	}
+	if metav1.HasAnnotation(etcd.ObjectMeta, annotationAllowUnhealthyPodEviction) {
```

Comment:
so, we are going to set `resources.druid.gardener.cloud/allow-unhealthy-pod-eviction` in `.metadata.annotation` but why not in `.spec.annotation` ? I'm asking this as CA `safe-to-evict` annotation for single node etcd-main is present in `.spec.annotation`

Reply:
Yes that is another possible place this annotation can be mentioned. However if you look closely then the `spec.annotations` in an `Etcd` resource is used in 2 places - STS component and compaction Job where it is set on the Job resource. If you set this annotation in `spec.annotations` then it also gets set on STS as well which does not make much sense.

Additionally this annotation should not be set directly as in on any resource managed by druid (which is the current purpose of annotations specified via `spec.annotations`). It is therefore correct to set it on etcd resource in `objectmeta.annotations` as it is a indication to druid that if this is set then on the PDB component set `pdb.Spec.UnhealthyPodEvictionPolicy` to `AlwaysAllow`.

Reply:
make sense. thanks